File: 2021/src/test/java/de/jfriemel/aoc/days/Day01Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day01Test {

    private static final String INPUT_STRING = """
            199
            200
            208
            210
            200
            207
            240
            269
            260
            263""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day01().part1(INPUT);
        assertEquals("7", result);
    }

    @Test
    public void testPart2() {
        String result = new Day01().part2(INPUT);
        assertEquals("5", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day01Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day01Test {
    private val input1 = """
        1abc2
        pqr3stu8vwx
        a1b2c3d4e5f
        treb7uchet
    """.trimIndent().lines()

    private val input2 = """
        two1nine
        eightwothree
        abcone2threexyz
        xtwone3four
        4nineeightseven2
        zoneight234
        7pqrstsixteen
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day01.part1(input1)
        Assertions.assertEquals("142", result)
    }

    @Test
    fun testPart2() {
        val result = Day01.part2(input2)
        Assertions.assertEquals("281", result)
    }
}
